src/box/sql.c

```diff
@@ -794,8 +794,10 @@ tarantoolsqlIdxKeyCompare(struct BtCursor *cursor,
 		}
 
 		if (fieldno != next_fieldno) {
-			struct tuple_field *field =
-				tuple_format_field(format, fieldno);
+			struct tuple_field *field = NULL;
+			if (fieldno < field_count)
+				field = tuple_format_field(format, fieldno);
+
 			if (fieldno >= field_count ||
```

Comment:
```suggestion
			if (field == NULL ||
```

Reply:
Thank you for the review and suggestion! Fixed.